packages/replay-internal/src/util/handleRecordingEmit.ts

```diff
@@ -59,38 +59,44 @@ export function getHandleRecordingEmit(replay: ReplayContainer): RecordingEmitCa
         return false;
       }
 
+      const session = replay.session;
+
       // Additionally, create a meta event that will capture certain SDK settings.
       // In order to handle buffer mode, this needs to either be done when we
-      // receive checkout events or at flush time.
+      // receive checkout events or at flush time. We have an experimental mode
+      // to perform multiple checkouts a session (the idea is to improve
+      // seeking during playback), so also only include if segmentId is 0.
       //
       // `isCheckout` is always true, but want to be explicit that it should
       // only be added for checkouts
-      addSettingsEvent(replay, isCheckout);
+      if (session && session.segmentId === 0) {
```

Comment:
I think we already guard this in the `addSettingsEvent` method? If we guard it here we can remove that check from there instead!

Reply:
Ahh, let's keep the guard in there, in case we end up calling `addSettingsEvent` elsewhere.